src/Container.ts

```diff
@@ -42,15 +42,16 @@ export class Container {
      *
      * @param name Service name. Optional
      * @param type Service class
-     * @param params Parameters to be sent to constructor on service initialization
+     * @param params Parameters to be sent to a class constructor or factory function on service initialization
+     * @param factory Factory function to be called on service initialization
      */
-    static registerService(name: string, type: Function, params?: any[]) {
-        this.registeredServices.push({ name: name, type: type, params: params });
+    static registerService(name: string, type: Function, params?: any[], factory?: Function) {
```

Comment:
probably we can rename this method to `register` and introduce an interface here to register service this way:

```ts
Container.register({
     name?: string,
     type: Function,
     params?: any[],
     factory?: (params?: any[]) => Function
});
```